pkg/controller/util/watch.go

```diff
@@ -0,0 +1,68 @@
+package util
+
+import (
+	"sync"
+	"time"
+
+	"github.com/openkruise/kruise/pkg/client"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/util/retry"
+	"k8s.io/klog/v2"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+)
+
+var watcherMap = sync.Map{}
+
+func DiscoverGVK(gvk schema.GroupVersionKind) bool {
```

Comment:
this func has been defined in `[pkg/util/discovery/discovery.go](https://github.com/openkruise/kruise/blob/14e21f5bcfd40790861e6db3dc75655c9c37b189/pkg/util/discovery/discovery.go)`